scripts/remote-dev-context7-device-login.py

```
@@ -0,0 +1,467 @@
+            label="Context7 login config directory",
+        )
+        context7_fd = open_owned_directory_at(
+            config_fd,
+            "context7",
+            expected_uid=expected_uid,
+            label="Context7 login credential directory",
+        )
+        try:
+            key_fd = os.open("credentials.json", key_flags, dir_fd=context7_fd)
+        except OSError as exc:
+            raise DeviceLoginError(f"Context7 login credentials are unavailable or unsafe: errno {exc.errno}") from exc
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)